pkg/controllers/provisioner/v1alpha1/controller.go

```diff
@@ -30,6 +31,8 @@ import (
 type Controller struct {
 	Client    client.Client
 	Allocator allocation.Allocator
+	sync.RWMutex
+	ProcessedPods map[string]bool
```

Comment:
I'd love to encapsulate all of this state (i.e. pending pods/nodes) inside a separate state object. We can isolate the mutexes there, too.

Reply:
Yes, I am going to create a cluster state object but this is to avoid processing same pod multiple times until I have that object ready.